conan/internal/api/install/generators.py

```diff
@@ -152,6 +156,28 @@ def _receive_conf(conanfile):
             conanfile.conf.compose_conf(build_require.conf_info)
 
 
+def _generate_sbom(conanfile):
+    cache_folder = conanfile._conan_helpers.home_folder, 
+    from conans.client.loader import load_python_file
+    sbom_plugin_path = HomePaths(cache_folder).sbom_manifest_plugin_path
+    if os.path.exists(sbom_plugin_path):
+        mod, _ = load_python_file(sbom_plugin_path)
+
+        if not hasattr(mod, "generate_sbom"):
+            raise ConanException(
+                f"SBOM manifest plugin does not have a 'generate_sbom' method")
+        if not callable(mod.generate_sbom):
+            raise ConanException(
+                f"SBOM manifest plugin 'generate_sbom' is not a function")
+
+        ConanOutput().warning(f"generating sbom")
+        # TODO: Think where it makes sense to generate the file at this point
+        safe_ref_filename = str(conan.ref).replace("/", "_").replace(".", "_")
+        outfile = os.path.join(os.path.curdir, f"sbom_{safe_ref_filename}.json") # Temp
```

Comment:
Let the extension code decide the final generated filename (or filenames), this shouldn't be an argument

Reply:
In other words, a user extension might decide to generate several different sboms/manifests files.